### PR TITLE
statement-distribution: prep for re-enabling

### PR DESCRIPTION
**Pull request by ordian**
In preparation for launching re-enabling (https://github.com/paritytech/polkadot-sdk/issues/2418), we need to adjust the disabling strategy of statement-distribution to use the relay parent's state instead of the latest state (union of active leaves). This will also ensure no raciness of getting the latest state vs accepting statements from disabling validators at the cost of being more lenient/potentially accepting more statements from disabled validators.

- [x] PRDoc

### CODE REVIEW

File: polkadot/node/network/statement-distribution/src/v2/mod.rs [hunk 1]
```diff
@@ -3333,12 +3295,16 @@ pub(crate) fn answer_request(state: &mut State, message: ResponderMessage) {
 		validated_in_group: !mask.validated_in_group.clone(),
 	};
 
-	// Ignore disabled validators from the latest state when sending the response.
-	let disabled_mask =
-		per_session.disabled_bitmask(group_index).expect("group existence checked; qed");
+	// Ignore disabled validators when sending the response.
+	let disabled_mask = relay_parent_state.disabled_bitmask(group);
```
Comment (ordian) [inline polkadot/node/network/statement-distribution/src/v2/mod.rs:3299]: not sure if this is needed anymore

Comment (Overkillus) [reply polkadot/node/network/statement-distribution/src/v2/mod.rs:3303]: Why not?

Correct me if I'm wrong but that means that a validator requesting a candidate within a context of a relay parent in which he is disabled will simply be ignored. Seems like a good property

**Review by Overkillus — COMMENTED** [review]
Thanks for prepping it :+1: Makes everything easier to reason about, and once we discuss the fragmentation vector LGTM

File: polkadot/node/network/statement-distribution/src/v2/mod.rs (hunk 1 above)
Comment (Overkillus) [reply polkadot/node/network/statement-distribution/src/v2/mod.rs:3303]: Discussed in sync - checks are already made when we receive requests. It was originally there due to raciness issues (consequence of using the leafs). Raciness should no longer be there with relay parent perspective so this extra check is unnecessary.

**Review by Overkillus — APPROVED** [review]

**Pull request merged**